src/components/elements/button/button.types.ts

```diff
@@ -0,0 +1,16 @@
+import type { Icon, IconWeight } from '@phosphor-icons/react';
+import type { AriaAttributes, AriaRole, HTMLAttributeAnchorTarget } from 'react';
+
+export interface ButtonProps extends AriaAttributes {
+  as?: 'button' | 'a' | 'Link';
```

Comment:
I think it is better to have just  `button`| `link`
```suggestion
  as?: 'button' | 'link';
```
And handle, which should be rendered in the component internally for the link, if isExternalUrl render a tag if no the Link component. wdyt?